components/uniset/src/props.rs

```diff
@@ -4,14 +4,13 @@
 
 use crate::enum_props::*;
 use crate::provider::*;
-use crate::{UnicodeSet, UnicodeSetError};
-use core::convert::TryInto;
+use crate::UnicodeSetError;
 use icu_provider::prelude::*;
 
-type UnisetResult = Result<UnicodeSet, UnicodeSetError>;
+type UnisetResult<'data> = Result<DataPayload<'data, UnicodePropertyV1Marker>, UnicodeSetError>;
 
 // helper fn
-fn get_prop<'data, D>(ppucd_provider: &D, resc_key: ResourceKey) -> UnisetResult
+fn get_prop<'data, D>(provider: &D, resc_key: ResourceKey) -> UnisetResult<'data>
```

Comment:
Praise: Thanks for eliminating the unnecessary `'data` from binary functions.

However, please continue investigating the same for the binary enumerated property functions.  As a backup, you could open a follow-up issue for this.

Reply:
Follow-up: #1095